inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -941,6 +941,39 @@ def normalize_collaborations(obj, eng):
     return obj
 
 
+def _get_correct_affiliation_data(affiliations):
+    cleaned_affiliations = []
+    for aff in affiliations:
+        cleaned_affiliations.append({key: val for key, val in aff.items() if key in ['value', 'record']})
+    return cleaned_affiliations
+
+
+def _extract_matched_aff_from_highlight(highlighted_raw_affs, author_raw_affs, author_affs):
+    highlight_matches = [re.finditer('(?=<em>).*?(?<=<\/em>)', raw_aff) for raw_aff in highlighted_raw_affs]
+    raw_aff_highlight_lenghts = [sum([match.span()[1] - match.span()[0] for match in aff_matches]) for aff_matches in highlight_matches]
```

Comment:
Instead of using lookaround and computing position differences, you can also improve the previous approach to use a capturing group.
If you define
```python
def _raw_aff_highlight_len(highlighted_raw_aff):
    matches = re.finditer(r"<em>(.*?)</em>", highlighted_raw_aff)
    return sum(len(match) for match in matches)
```
you can do
```suggestion
    raw_aff_highlight_lenghts = [_raw_aff_highlight_len(raw_aff) for raw_aff in highlighted_raw_affs]
```